### ISSUE TITLE
win32 ONNX release

### ISSUE THREAD

**Issue by adityagoel4512** [onnx/onnx#5448]
I wanted to copy over an issue that emerged in https://github.com/onnx/onnx/pull/5401.
 
TLDR: are win32 releases a priority for ONNX? It looks like 32 bit Linux support has been dropped so I wanted to double check. Thanks.
              
              Slight issue here: google-re2 and pyre2 don't ship win32 wheels and one of these needs to be a dependency for the reference implementation of RegexFullMatch (it is not used anywhere else). This is causing the windows release CI to fail as it attempts to build pyre2 from the source distribution which is non trivial.
              I had a [look](https://github.com/adityagoel4512/pyre2/actions/runs/5680016161/job/15393282432#step:6:293) at it and it should in principle be possible to build win32 wheels. Before going any further I wanted to first check to see if win32 is actively being supported by ONNX or if it is unintentionally present in this CI? It looks like 32 bit Linux is dropped so I was not sure.

Should I create a separate issue for this?

_Originally posted by @adityagoel4512 in https://github.com/onnx/onnx/issues/5401#issuecomment-1653490632_
